oat-sa/extension-tao-delivery
# Feature/tca 651/login queue ce

Login queue capacity restriction moved to tao delivery.
 
Related to : https://oat-sa.atlassian.net/browse/TCA-651
  
#### How to test
 

- Configure `oat\tao\model\actionQueue\ActionQueue::SERVICE_ID` with `oat\ltiDeliveryProvider\model\actions\GetActiveDeliveryExecution` action, which would have the new `oat\taoDelivery\model\Capacity\CapacityRestriction`. f.e.:

```
return new oat\tao\model\actionQueue\implementation\InstantActionQueue(array(
    'persistence' => 'redis',
    'actions' => array(
        'oat\\ltiDeliveryProvider\\model\\actions\\GetActiveDeliveryExecution' => array(
            'ttl' => 60,
            'restrictions' => array(
                'oat\\taoDelivery\\model\\Restriction\\CapacityRestriction' => 1
            )
        )
    )
));
```
- Check that `oat\taoDelivery\model\Capacity\CapacityInterface::SERVICE_ID` is configured as well. f.e.:
```
return new oat\taoDelivery\model\Capacity\DummyCapacityService(array(
    'capacity' => 1,
));
```
- Launch an LTI test, test that when `DummyCapacityService` is configured with `capacity == 0` then login queue waiting screen is presented, and upon changing the `capacity` to `1` LTI test starts without login queue interfering. 
 
Test commands :
 
```
./vendor/bin/phpunit taoDelivery/test/unit/model/Restriction/
```
 
#### Dependencies
 
Requires :
 - [ ] https://github.com/oat-sa/tao-core/pull/2553

Companion PR :
 - [ ] https://github.com/oat-sa/extension-tao-ltideliveryprovider/pull/254
 - [ ] https://github.com/oat-sa/extension-tao-act/pull/1396
